python/pyiceberg/expressions/base.py

```diff
@@ -169,16 +171,64 @@ def bind(self, schema: Schema, case_sensitive: bool) -> BoundReference[T]:
         return BoundReference(field=field, accessor=accessor)
 
 
-@dataclass(frozen=True)  # type: ignore[misc]
 class BoundPredicate(Bound[T], BooleanExpression):
```

Comment:
For readability and type checking it is best to also define the variables in the class itself:
```python
class BoundPredicate(Bound[T], BooleanExpression):
    _term: BoundTerm[T]
    _literals: Tuple[Literal[T]]
```
This will also remove the ignore below:
```python
@property
def literals(self) -> tuple[Literal[T], ...]:
    return self._literals  # type: ignore
```

Reply:
Done!